### DISCUSSION TITLE
Where can we learn more about/prepare for 3.0?

### DISCUSSION THREAD

**Discussion by erikdstock** [saadeghi/daisyui#1747] [General]
I'm embarking on a new project and evaluating daisyUI, but with the new '3.0 coming soon' badge it would be nice to plan for that, knowing whether a big migration will be necessary, etc. Is there any publicly-available info about this that i'm missing?

**Comment by saadeghi** [comment]
I'm still working on it. It's on `v3` branch.  
I haven't documented the changelogs yet but you can check the commit messages to see what's changed.  

https://github.com/saadeghi/daisyui/commits/v3

There will be more changes. I'm going through all issues and discussions to fix anything that I couldn't do in a minor releases.   

Regarding the update, Ideally, you might need to change a few class names and a few HTML tags. Not that much.  
I will provide a complete guide on what exactly should be changed in the HTML so don't worry about that now. You can use v2 and when v3 is ready, it would take maximum 2 or 3 mins to change things based on the guide I will provide.

**Reply by dionysuzx** [reply]
out of curiosity is there going to be any improvements to the accessibility on modals? i believe that was the main area where i was getting lint errors for svelte about accessibility. if i remember correctly, you would need to click out of the modal area and it didn't support "escape" character for example. i was going to try to wrap daisyUI components in headless UI components in order to achieve the accessibility.

and second question when is the expected release of v3? thanks!

**Reply by saadeghi** [reply]
>is there going to be any improvements to the accessibility on modals?  

Yes. We will be using the new `<dialog>` element for modals which fixes the accessibility concerns.  

>I was going to try to wrap daisyUI components in headless UI components in order to achieve the accessibility

That's a good practice anyway. daisyUI is just the CSS and it's aimed to provide visual styles. Headless UI (or any headless JS library) aim to provide functionality without (or with optional) visual styles and design decisions. So they are a good match together.

> when is the expected release of v3?  

Next week

**Comment by rolanday** [comment]
+1 , love Daisy UI css-only approach, and the sematic colors & theming!! Thank you so much for the great work. Now stop teasing up w/ 3.0 billboard message :-) Seriously though, thank you again.

edit: one more thing, as feat request, would be nice to have more fine-grained controller for some of the UI elements in config file, button sizes in particular -- I find the small a little too small and md a little too big, so right now just disabling styles, but would be nice to keep them enabled.

**Reply by saadeghi** [reply]
**About button sizes**:  
There are many ways to customize them:  
You can use `min-h-0` and `h-*` utility classes to remove the min height and use your own height (either for 1 button or `@apply` it for all `.btn` classes)  
You can customize font size or width as well (or any other CSS rule).
You can also write custom styles for a specific theme in the config.  

The thing is, button sizes are not just about buttons. They should look good with other elements as well so ideally we want all data input elements and data display elements look good together when we use them next to each other.

Let me know if you have any questions.

**Comment by NayamAmarshe** [comment]
Are there new themes? I would love to see a a more modern and glassy theme similar to [NextUI](https://nextui.org/)

**Reply by saadeghi** [reply]
This update is focused on bug fixes, accessibility improvements and new features.  
After the release I will be working on more themes but I want to encourage devs to create their own unique theme for their projects. With [theme generator](https://daisyui.com/theme-generator/) you can easily pick your brand colors and make a new theme quickly. Then you can even customize more (border-radius, shadows, etc) in the theme config [using CSS variables](https://daisyui.com/docs/themes/#-5) or [any custom CSS](https://daisyui.com/docs/themes/#-8)  

So basically you can make a new theme that looks like Next UI in less than a minute.  
If by "glassy theme" you mean [these buttons](https://nextui.org/docs/components/button#shadow), You can do it using Tailwind's `shadow-*` class: https://play.tailwindcss.com/Wga1GwxnNJ

Let me know if you have any questions
